File: code/community/1298601623001501697/deepgram-medical-model-mvp.js
const Deepgram = require('@deepgram/sdk');
const { Readable } = require('stream');
require('dotenv').config();

const deepgramApiKey = process.env.DEEPGRAM_API_KEY;
const model = 'nova-2-medical';

const deepgram = new Deepgram(deepgramApiKey);

const audioSource = '<YOUR_AUDIO_SOURCE_URL>'; // URL to the audio file

// Function to transcribe audio
async function transcribe() {
    try {
        const response = await deepgram.transcription.preRecorded({
            url: audioSource
        }, {
            model: model,
            language: 'en-US'
        });

        console.log('Transcript:', response.results.channels[0].alternatives[0].transcript);
    } catch (error) {
        console.error('Error transcribing audio:', error);
    }
}

transcribe();

File: code/community/1298601623001501697/deepgram-medical-model-mvp.cs
using Deepgram;
using Deepgram.Transcription;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

class Program
{
    static async Task Main(string[] args)
    {
        var apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        var model = "nova-2-medical";

        var deepgramClient = new DeepgramClient(apiKey);

        var audioSource = new Dictionary<string, string> {
            { "url", "<YOUR_AUDIO_SOURCE_URL>" }  // Replace with your audio source
        };

        var options = new TranscriptionOptions()
        {
            Model = model,
            Language = "en-US"
        };

        var response = await deepgramClient.Transcription.PreRecordedAsync(audioSource, options);

        Console.WriteLine("Transcript: " + response.Results.Channels[0].Alternatives[0].Transcript);
    }
}